app/models/ZuoraCatalog/ZuoraCatalog.scala

```diff
@@ -0,0 +1,72 @@
+package models.ZuoraCatalog
+
+case class ZuoraCatalogPricing(
+    currency: String,
+    price: Double
+)
+
+case class ZuoraCatalogCharge(
+    endDateCondition: Option[String],
+    pricing: List[ZuoraCatalogPricing]
+)
+
+case class ZuoraCatalogProductRatePlan(
+    name: Option[String],
+    id: String,
+    productRatePlanCharges: List[ZuoraCatalogCharge]
+) {
+  def price: Double = {
+    productRatePlanCharges
+      .filter(_.endDateCondition.contains("Subscription_End"))
+      .map(
+        _.pricing
+          .filter(_.currency == "GBP")
```

Comment:
I don't understand why there's a restriction to GBP?

Reply:
is `pricing` always an array of currencies? right now I'm adding all GBP ones, but if it's always 1 price per currency I can map it to the final result